web3/tools/benchmark/main.py

```diff
@@ -144,18 +144,20 @@ def main(logger: logging.Logger, num_calls: int) -> None:
                 {
                     "name": "eth_blockNumber",
                     "params": {},
-                    "exec": lambda: w3_http.eth.block_number,
-                    "async_exec": lambda: async_w3_http.eth.block_number,
+                    "exec": lambda w3_http=w3_http: w3_http.eth.block_number,
+                    "async_exec": lambda async_w3_http=async_w3_http: async_w3_http.eth.block_number,  # noqa: E501
                 },
                 {
                     "name": "eth_getBlock",
                     "params": {},
-                    "exec": lambda: w3_http.eth.get_block(1),
-                    "async_exec": lambda: async_w3_http.eth.get_block(1),
+                    "exec": lambda w3_http=w3_http: w3_http.eth.get_block(1),
+                    "async_exec": lambda async_w3_http=async_w3_http: async_w3_http.eth.get_block(  # noqa: E501
+                        1
+                    ),
                 },
             ]
 
-            def benchmark(method: Dict[str, Any]) -> None:
+            def benchmark(method: Dict[str, Any], loop=loop) -> None:
```

Comment:
The error is:

`B023 Function definition does not bind loop variable 'loop'.`

It's because `loop` and the function are defined within a `for` loop, the `loop=loop` fastens the value of `loop` to the function each time it's created, rather than just the value of `loop` at the end of the `for`. Or something like that.

Reply:
Got it. Thanks for the clarification!